examples/runners/wgpu/src/compute.rs

```diff
@@ -97,8 +107,33 @@ pub fn start(options: &Options) {
         let mut cpass = encoder.begin_compute_pass(&wgpu::ComputePassDescriptor { label: None });
         cpass.set_bind_group(0, &bind_group, &[]);
         cpass.set_pipeline(&compute_pipeline);
-        cpass.dispatch(1, 1, 1);
+        cpass.dispatch(src.len() as u32, 1, 1);
     }
 
+    encoder.copy_buffer_to_buffer(&storage_buffer, 0, &staging_buffer, 0, size);
     queue.submit(Some(encoder.finish()));
+    let buffer_slice = staging_buffer.slice(..);
+    let buffer_future = buffer_slice.map_async(wgpu::MapMode::Read);
+    device.poll(wgpu::Maintain::Wait);
+    if let Ok(()) = {
+        cfg_if::cfg_if! {
```

Comment:
this feels a bit hard to read. Maybe extract this into a binding above like `let buffer_readback_result = ...`